src/gpuarray_collectives_cuda_nccl.c

```diff
@@ -28,7 +28,7 @@ static const char* nccl_success_error = ncclGetErrorString(ncclSuccess);
       (ctx)->error_msg = ncclGetErrorString(nccl_err); \
       return GA_COMM_ERROR;                            \
     }                                                  \
-    (ctx)->error_msg = nccl_success_error;             \
```

Comment:
I would say that this is unnecessary.  We can leave whatever was there before if there is no error.